src/main/java/com/SAS/LeagueManagement/LeagueManagementController.java

```diff
@@ -0,0 +1,226 @@
+     */
+    public LeagueManagementController() {
+        this.rankPolicies = new LinkedList<>();
+        this.pointsPolicies = new LinkedList<>();
+        this.gamesPolicies = new LinkedList<>();
+        initPolicies();
+    }
+
+    /**
+     * The functions initialize the available policies in our system
+     */
+    private void initPolicies() {
+        //add rank policies
+        rankPolicies.add(new NumberOfWins());
+        rankPolicies.add(new GoalDifference());
+
+        //add points policies
+        pointsPolicies.add(new OnePointForWinAndNoneForDraw());
+        pointsPolicies.add(new TwoForWinOneForDraw());
+        pointsPolicies.add(new ThreeForWinOneForDrawPolicy());
+
+        //add games policies
+        gamesPolicies.add(new OneRoundLeague());
+        gamesPolicies.add(new TwoRoundsLeague());
+        gamesPolicies.add(new ThreeRoundsLeague());
+    }
+
+    /**
+     * The function returns all the available rank policies in our system
+     * @return
+     */
+    public String showRankPolicies(){
+        int counter = 1;
+        StringBuilder policies = new StringBuilder();
+
+        for (LeagueRankPolicy policy : rankPolicies) {
+            policies.append(counter + ". " + policy.getName() + "\n");
+        }
+
+        policies.setLength(policies.length() - 1);
+        return policies.toString();
+    }
+
+    /**
+     * The function returns all the available points policies in our system
+     * @return
+     */
+    public String showPointsPolicies(){
+        int counter = 1;
+        StringBuilder policies = new StringBuilder();
+
+        for (PointsPolicy policy : pointsPolicies) {
+            policies.append(counter + ". " + policy.getName() + "\n");
+        }
+
+        policies.setLength(policies.length() - 1);
+        return policies.toString();
+    }
+
+    /**
+     * The function returns all the available game policies in our system
+     * @return
+     */
+    public String showGamePolicies(){
+        int counter = 1;
+        StringBuilder policies = new StringBuilder();
+
+        for (GamesPolicy policy : gamesPolicies) {
+            policies.append(counter + ". " + policy.getName() + "\n");
+        }
+
+        policies.setLength(policies.length() - 1);
+        return policies.toString();
+    }
+
+    /**
+     * The function receives league, season and rank policy id and set the rank policy to the league according to the season
+     * @param league
+     * @param season
+     * @param rankPolicyId
+     * @return
+     */
+    public boolean addRankPolicy(League league, Season season, String rankPolicyId){
+        if (league == null || season == null) {
+            return false;
+        }
+
+        if (rankPolicyId.length() == 0) {
+            return false;
+        }
+
+        LeagueRankPolicy rankPolicy;
+
+        //TODO: check permissions
+        switch (rankPolicyId) {
+            case "1":
+                rankPolicy = new NumberOfWins(league,season);
```

Comment:
I think you can can the two rows and put them after the switch case.
league.addRankPolicy(season,rankPolicy);
season.addRankPolicy(league,rankPolicy);

Reply:
agree